types/src/state_store/state_key/registry.rs

```diff
@@ -152,7 +152,7 @@ where
         let mut locked = self.inner.write();
         if let Some(map2) = locked.get_mut(key1) {
             if let Some(entry) = map2.get(key2) {
-                if entry.upgrade().is_none() {
+                if entry.strong_count() == 0 {
                     map2.remove(key2);
```

Comment:
Why not move the other conditional inside this `if` statement, while already here :)

Reply:
sure. I tried to think of why I wrote it that way other than one less level of nesting but I can't 🧌